### PR TITLE
feat: create custom payment method

### PR DESCRIPTION
**Pull request by frobel**
- add possibility to create payment method
- other refactorings

### CODE REVIEW

File: src/services/TestDataService.ts [hunk 1]
```diff
@@ -1255,13 +1362,14 @@ export class TestDataService {
...
-        if (orderDelivery.shippingCosts != null) {
-            shippingCosts = orderDelivery.shippingCosts.totalPrice;
-        }
+        const orderDelivery = this.getBasicOrderDeliveryStruct(deliveryState, shippingMethod, customerAddress);
+        orderDelivery.shippingCosts ? shippingCosts = orderDelivery.shippingCosts.totalPrice : shippingCosts = 0;
+
         totalPrice += shippingCosts;
 
+        const orderTransaction = this.getBasicOrderTransactionStruct(paymentMethod, transactionState);
+        orderTransaction.amount ? orderTransaction.amount.totalPrice = totalPrice : totalPrice;
```
Comment (frobel) [inline src/services/TestDataService.ts:1371]: @Phil23 is that possible to define somehow that `orderTransactions.amount` could not be undefined?

File: src/services/TestDataService.ts [hunk 2]
```diff
@@ -1255,13 +1362,14 @@ export class TestDataService {
                 }
             }
         });
-        const orderDelivery = this.getBasicOrderDeliveryStruct(deliveryState, shippingMethod, customerAddress)
-        let shippingCosts = 0;
-        if (orderDelivery.shippingCosts != null) {
-            shippingCosts = orderDelivery.shippingCosts.totalPrice;
-        }
+        const orderDelivery = this.getBasicOrderDeliveryStruct(deliveryState, shippingMethod, customerAddress);
+        orderDelivery.shippingCosts ? shippingCosts = orderDelivery.shippingCosts.totalPrice : shippingCosts = 0;
```
Comment (frobel) [inline src/services/TestDataService.ts]: @Phil23 is that possible to define somehow that `orderDelivery.shippingCosts` could not be undefined?

Comment (SebastianFranze) [reply src/services/TestDataService.ts]: This should be easier to read:

```suggestion
        const shippingCosts = orderDelivery.shippingCosts?.totalPrice ?? 0;
```

File: src/services/TestDataService.ts [hunk 3]
```diff
@@ -1230,6 +1304,7 @@ export class TestDataService {
         overrides: Partial<Order> = {},
     ): Partial<Order> {
 
+        let shippingCosts;
```
Comment (SebastianFranze) [inline src/services/TestDataService.ts]:
```suggestion
```

**Review by SebastianFranze — APPROVED** [review]

**Pull request merged**